src/clusterfuzz/_internal/bot/tasks/utasks/__init__.py

```diff
@@ -138,6 +138,11 @@ def __exit__(self, _exc_type, _exc_value, _traceback):
     monitoring_metrics.UTASK_SUBTASK_E2E_DURATION_SECS.add(
         e2e_duration_secs, self._labels)
 
+    outcome = 'error' if _exc_type else 'success'
```

Comment:
Is there any possible situation where we return an error code as a value, and not throw an exception, when a task fails in any of the pre/main/post steps?

Reply:
https://github.com/google/clusterfuzz/blob/089d9ef1a8f6bdb03a9046ad5c3286874384cf3a/src/clusterfuzz/_internal/protos/uworker_msg.proto#L355

Should probably be introspecting the output to pinpoint failure if the result is anything other than NO_ERROR?

Reply:
Can't really think of any other scenario where a task failure does not imply an exception.